### ISSUE TITLE
`invalid-argument-type` when variable keyword function parameter is defined

### ISSUE THREAD

**Issue by dmitry-gils** [astral-sh/ty#2583]
### Summary

Example is minimal and self-explanatory: https://play.ty.dev/550d7622-7950-4623-803d-3301aeb55808

```py
from typing import Any

def fn(**kwargs: dict[str, Any]):
    ...

fn(param=[1,2,3])
```

Linter is expected to 'unpack' `**kwargs` type definition and validate `param=[1,2,3]` against `str` and `list`, not `dict`.

### Version

0.0.13

**Comment by AlexWaygood** [comment]
Hey, thanks for the report! I'm afraid ty is correct here, however. [Mypy](https://mypy-play.net/?mypy=latest&python=3.12&gist=48dbf060b4dd07edd6dc4916c3e2ed20), [pyright](https://pyright-play.net/?pyrightVersion=1.1.405&pythonVersion=3.13&reportUnreachable=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoCCKcAUCQCYCmwUwKAFAFSMDWA7gIYhoDOAXFHJIAxjADaPGCAA0hYgF0AlHxJQ1UAHRaydegi4cIAXjEBGaQCZpAZiVA) and [pyrefly](https://pyrefly.org/sandbox/?project=N4IgZglgNgpgziAXKOBDAdgEwEYHsAeAdAA4CeS4ATrgLYAEALqcROgOZ0Q3G6UN0BBdKQA66MZhhg6YdAAoAVAoDWAd1SU2cRHUwQAxgwDacBpQA0g4QF0AlIjF0ndQq7FjZc4htQ0AvEYAjOYATOYAzHZiIOYgAK4M0HAk5IggAMR0AKqJUBBMMnHohhC46HDuWFIyvDSoDAD66HE02DCUcvg6rAy2dAC0AHx0ppQO6M50lDAMcZQTYCIgAHItbWN0wPgAvkvRsWTTYFCkhAy0UBSZAAqkRycjGDgEdPplkGxz9aXohGKZAGUYDA6AALBgMYjaAD00MOUhOhF4bGhMHQ0MwuH0cGhb3QHy%2BiTK0JqlDoqAAbqhoKhsLBXu8IJ9KN8ynRcMQieU-ugyAxQWV%2BhT2nAfnQ-HQluFCIEQnt0CBtrFUCVhQAxaAwChoLB4IhkRVAA) all agree with us, because this is clearly specified by the typing spec here: https://typing.python.org/en/latest/spec/callables.html#annotating-args-and-kwargs
